File: macos/Sources/Models/LogEntry.swift
import Foundation

enum LogEventType: String {
  case deviceConnected
  case deviceDisconnected
  case pairingComplete
  case pairingFailed
}

enum LogDirection {
  case incoming
  case outgoing
  case local
}

enum LogStatus {
  case success
  case failed
}

struct LogEntry: Identifiable {
  private static let timeFormatter: DateFormatter = {
    let fmt = DateFormatter()
    fmt.dateFormat = "HH:mm"
    return fmt
  }()

  let id = UUID()
  let timestamp: Date
  let type: LogEventType
  let message: String
  let direction: LogDirection
  let status: LogStatus
  let error: String?

  init(
    timestamp: Date = Date(), type: LogEventType, message: String,
    direction: LogDirection = .local, status: LogStatus = .success, error: String? = nil
  ) {
    self.timestamp = timestamp
    self.type = type
    self.message = message
    self.direction = direction
    self.status = status
    self.error = error
  }

  var timeString: String { Self.timeFormatter.string(from: timestamp) }
}

File: android/app/src/main/java/com/buzzel/model/LogEntry.kt
package com.buzzel.model

import java.text.SimpleDateFormat
import java.util.Date
import java.util.Locale

enum class LogEventType {
    DEVICE_CONNECTED,
    DEVICE_DISCONNECTED,
    PAIRING_STARTED,
    PAIRING_COMPLETE,
    PAIRING_FAILED,
    REMOTE_FOCUS,
    REMOTE_BLUR,
}

enum class LogDirection {
    INCOMING,
    OUTGOING,
    LOCAL,
}

enum class LogStatus {
    SUCCESS,
    FAILED,
}

data class LogEntry(
    val timestamp: Long = System.currentTimeMillis(),
    val type: LogEventType,
    val message: String,
    val direction: LogDirection = LogDirection.LOCAL,
    val status: LogStatus = LogStatus.SUCCESS,
    val error: String? = null,
) {
    val timeString: String
        get() = TIME_FORMAT.format(Date(timestamp))

    companion object {
        private val TIME_FORMAT = SimpleDateFormat("HH:mm", Locale.getDefault())
    }
}
